lib/stream_abstract.php

```diff
@@ -36,6 +36,29 @@ public function setTitle($value)
         $this->title = $value;
     }
 
+	/**
+	 * Get in YFeed database stored items belonging to this stream orderd by date.
+	 * @param int $number Number of items to be returned
+	 * @return \rex_yfeed_item[] Array with item objects
+	 */
+	public function getPreloadedItems($number = 5)
+	{
+		$items = [];
+		$result = rex_sql::factory();
+		$result->setQuery('SELECT id FROM '. rex::getTablePrefix() .'yfeed_item ORDER BY updatedate DESC LIMIT 0, '. $number .';');
+
+		for ($i = 0; $i < $result->getRows(); $i++) {
+			$items[] = rex_yfeed_item::get($result->getValue('id'));
```

Comment:
Hier wäre es besser, wenn direkt die Daten aus der einen Query oben kommen würde, und nicht für jedes Item noch mal separat die Daten abgefragt werden müssten.

Reply:
Das wäre bestimmt performanter. Aber:
- Von der Logik her gehört diese Methode in die rex_yfeed_stream_abstract Klasse. Außerdem wäre die Trennung der Logik verletzt, wenn wir hier rex_yfeed_item Instanzen erstellen.
- Wegen der Sichtbarkeit der Variablen der rex_yfeed_item Klasse können hier keine Werte zugewiesen werden.

Wenn du möchtest, können wir aber in der rex_yfeed_item Klasse eine Methode getPreloadedStreamItems($streamId, $numberItems = 5) erstellen.